### ISSUE TITLE
CDK Pipeline is not triggered

### ISSUE THREAD

**Issue by asadpiz** [awslabs/aws-service-catalog-factory#346]
Please ensure you are using the latest version and have tried to run a validate command on your source files. 

I'm unable to create a CDK product pipeline using the latest SCF. The **servicecatalog-factory-pipeline** runs successfully but it creates the default CloudFormation pipeline instead of the CDK one, which fails with the following error:

```The user-provided path product.template.yaml does not exist.```

# Steps to reproduce
1. I'm trying to create a CDK deployment product using the following product yaml file:

```yaml
Schema: factory-2019-04-01
Products:
  - Name: "aws-sct-cdk-product-1"
    Owner: "aws-sct@test.com"
    Description: "aws-sct-product"
    Distributor: "cloud-engineering"
    SupportDescription: "aws-sct-cdk-product-1"
    SupportEmail: "aws-sct@test.com"
    SupportUrl: "https://test.com"
    Tags:
      - Key: "name"
        Value: "aws-sct-cdk-product-1"
    Versions:
      - Name: "v1"
        Description: "v1 of aws-sct-cdk-product-1"
        Template:
          Name: "CDK"
          Version: "1.0.0"
          Configuration:
            runtime-versions:
              python: "3.10"
            install:
              commands:
                - "pip install -r requirements.txt"
        Source:
          Provider: "CodeStarSourceConnection"
          Configuration:
            BranchName: "main"
            ConnectionArn: "arn:aws:codestar-connections:xxxxxxx"
            FullRepositoryId: "OrgID/RepoCode"
            OutputArtifactFormat: "CODE_ZIP"
```
1. The product pipeline fails at the **Test stage** with the error ```The user-provided path product.template.yaml does not exist.```
1. 

# Expected results
1. The special product pipeline for CDK is created instead of the CloudFormation one. 

# Actual results
The product pipeline fails at the **Test stage** with the error ```The user-provided path product.template.yaml does not exist.```

**Comment by asadpiz** [comment]
I'd also like to add that when using CodeStarSourceConnection, I have to use the configuration parameter ```OutputArtifactFormat```, otherwise the **servicecatalog-factory-pipeline** fails.

**Comment by eamonnfaherty** [comment]
the solution is generating a generic-combined-pipeline task for the product version - it is ignoring the template attribute.

**Comment by eamonnfaherty** [comment]
Within CreateGenericCombinedPipelineTask item and versions both contain the template details.

**Comment by asadpiz** [comment]
Fixed in [0.102.0](https://github.com/awslabs/aws-service-catalog-factory/releases/tag/0.102.0)
